src/main/java/Duke.java

```diff
@@ -1,10 +1,133 @@
+import java.util.*;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Linus!\nWhat can I do for you?\n\n");
+
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner sc = new Scanner(System.in);
+
+        // while loop to repeat printing of multiple Scanner inputs
+        // adapted with help of AI
+        while (true) {
+            String input = sc.nextLine();
+
+            try {
+                // Check if the user wants to exit.
+                // When comparing strings for equality, you should use the equals() method, not the == operator.
+                if (input.equals("bye")) { // exit chat
+                    System.out.println("Bye. It's been a pleasure chatting with you!");
+                    break;
+                } else if (input.equals("list")) { // list tasks
+                    System.out.println("Here are the tasks in your list:");
+
+                    for (int i = 0; i < tasks.size(); i++) {
+                        Task currTask = tasks.get(i);
+                        System.out.println((i + 1) + ". " + currTask);
+                    }
+                } else if (input.startsWith("mark")) {
+                    Integer indexOfTask = Integer.parseInt(input.substring(5));
+                    Task currTask = tasks.get(indexOfTask);
+                    currTask.markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(currTask);
+                } else if (input.startsWith("unmark")) {
+                    Integer indexOfTask = Integer.parseInt(input.substring(7));
+                    Task currTask = tasks.get(indexOfTask);
+                    currTask.markAsNotDone();
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.println(currTask);
+                } else if (input.startsWith("todo")) {
+                    // Check if the input string is long enough
+                    if (input.length() <= 5) {
+                        throw new DukeException("Please specify the description of the todo task, " +
+                                "starting from one whitespace away from the keyword 'todo'" +
```

Comment:
Really good practice of the correct indentation here. It made string message so easy to read.

Reply:
I agree with this. Correct indentation for "if else" has helped with readability for the logic of the code too.